src/ray/core_worker/transport/direct_actor_transport.h

```diff
@@ -244,12 +239,17 @@ class SchedulingQueue {
     pending_tasks_[seq_no] =
         InboundRequest(accept_request, reject_request, dependencies.size() > 0);
     if (dependencies.size() > 0) {
-      waiter_.Wait(dependencies, [seq_no, this]() {
+      waiter_.Wait(dependencies, [seq_no, this](Status status) {
         RAY_CHECK(boost::this_thread::get_id() == main_thread_id_);
         auto it = pending_tasks_.find(seq_no);
         if (it != pending_tasks_.end()) {
           it->second.MarkDependenciesSatisfied();
-          ScheduleRequests();
+          if (!status.ok()) {
+            it->second.Cancel();
+            pending_tasks_.erase(it);
+          } else {
+            ScheduleRequests();
```

Comment:
Can we just call ScheduleRequests() in all cases? The requests will fail by themselves.

I'm pretty sure the cancellation path above is also incomplete.

Reply:
How would they fail by themselves? I think we still need to cancel individual tasks. But yeah, I'll put the ScheduleRequests() call back and add it to the unit test.

Reply:
They will execute, call ray.get() on the arg, and that will fail. There is no need to cancel them.

Reply:
Oh I see, the task will just fail if the arguments aren't local? I like that better too, will revert.